docs/test.ts

```diff
@@ -116,7 +116,13 @@ const graphqlOptions = {
   depth: 2,
   variables: ({ foo, baz }) => ({ foo }),
   collapse: false,
-  signature: false
+  signature: false,
+  hooks: {
+    parse: (span, args) => {},
+    validate: (span, args) => {},
+    execute: (span, args, res) => {},
```

Comment:
Given the fundamental differences between these different hooks, I'd say to focus this PR only on `execute` for now and discuss the others separately.

Reply:
Makes sense. I reduced the scope in https://github.com/DataDog/dd-trace-js/pull/909/commits/e3edb8fb355820ca2bf2ee419cdbc75b59038315.